### PR TITLE
Moved more things to debug filters, and associated cleanup

### PR DESCRIPTION
**Pull request by EatThePath**
This is a number of items that to me practically never have general log value and hide things that do have value. So, if anything shouldn't be hidden away like this, by all means say so

### CODE REVIEW

**Comment by wookieejedi** [comment]
I use the parsing model and sub model (and which is detail level) along with converting sounds myself, though not sure how common that is.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Since you're doing associated cleanup, you might want to change your "nullptr" for -luadev.

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Also your `nprintf` commands in `ADE_FUNC(warning` and `ADE_FUNC(error` need newlines.

File: code/bmpman/bmpman.cpp [hunk 1]
```diff
@@ -1272,7 +1272,7 @@ bool bm_load_and_parse_eff(const char *filename, int dir_type, int *nframes, int
 	}
 	catch (const parse::ParseException& e)
 	{
-		mprintf(("BMPMAN: Unable to parse '%s'!  Error message = %s.\n", filename, e.what()));
+		nprintf(("BpMan","BMPMAN: Unable to parse '%s'!  Error message = %s.\n", filename, e.what()));
```
Comment (Goober5000) [inline code/bmpman/bmpman.cpp:1275]: This abbreviation should be "BmpMan", not "BpMan", surely.

Though backing up a bit, if these BmpMan messages are legit errors, why hide them?

File: code/bmpman/bmpman.cpp [hunk 2]
```diff
@@ -1526,12 +1526,12 @@ int bm_load_animation(const char *real_filename, int *nframes, int *fps, int *ke
...
 	if (type == BM_TYPE_EFF) {
 		if (!bm_load_and_parse_eff(filename, dir_type, &anim_frames, &anim_fps, &key, &eff_type)) {
-			mprintf(("BMPMAN: Error reading EFF\n"));
+			nprintf(("BpMan","BMPMAN: Error reading EFF\n"));
 			if (img_cfp != nullptr)
 				cfclose(img_cfp);
 			return -1;
 		} else {
-			mprintf(("BMPMAN: Found EFF (%s) with %d frames at %d fps.\n", filename, anim_frames, anim_fps));
+			nprintf(("BMPMAN: Found EFF (%s) with %d frames at %d fps.\n", filename, anim_frames, anim_fps));
```
Comment (Goober5000) [inline code/bmpman/bmpman.cpp:1534]: This `nprintf` is missing a category

File: code/model/modelread.cpp [hunk 3]
```diff
@@ -3211,7 +3211,7 @@ int model_load(const  char* filename, int n_subsystems, model_subsystem* subsyst
 
 	TRACE_SCOPE(tracing::LoadModelFile);
 
-	mprintf(( "Loading model '%s' into slot '%i'\n", filename, num ));
+	nprintf(("Model" "Loading model '%s' into slot '%i'\n", filename, num ));
```
Comment (Goober5000) [inline code/model/modelread.cpp:3214]: This `nprintf` is missing a comma, which effectively means it's missing a category

File: code/model/modelread.cpp [hunk 4]
```diff
@@ -3396,7 +3396,7 @@ int model_load(const  char* filename, int n_subsystems, model_subsystem* subsyst
 					dl2--;	// Start from 1 up...
 					if (dl2 >= sm1->num_details ) sm1->num_details = dl2+1;
 					sm1->details[dl2] = j;
-  				    mprintf(( "Submodel '%s' is detail level %d of '%s'\n", sm2->name, dl2 + 1, sm1->name ));
+  				    nprintf(("Model" "Submodel '%s' is detail level %d of '%s'\n", sm2->name, dl2 + 1, sm1->name ));
```
Comment (Goober5000) [inline code/model/modelread.cpp:3399]: This one too

File: code/model/modelread.cpp (hunk 3 above)
Comment (EatThePath) [reply code/model/modelread.cpp:3214]: I'm shocked that compiles honestly

**Comment by EatThePath** [comment]
> I use the parsing model and sub model (and which is detail level) along with converting sounds myself, though not sure how common that is.

if it matters, I didn't touch any messages about converting sounds in general, just the retail numbered sounds.

**Comment by wookieejedi** [comment]
> > I use the parsing model and sub model (and which is detail level) along with converting sounds myself, though not sure how common that is.
> 
> if it matters, I didn't touch any messages about converting sounds in general, just the retail numbered sounds.

Ah that makes sense, thanks!

**Review by Goober5000 — COMMENTED** [review]
Looks mostly good now, but my question remains: Are some of these legit errors?  If they are errors, not just spurious messages, they should be kept.

**Comment by EatThePath** [comment]
> Looks mostly good now, but my question remains: Are some of these legit errors? If they are errors, not just spurious messages, they should be kept.

I agree. I may have been a bit hasty in places, I'll do my best to give it another look through over the weekend and move anything serious back out.

File: code/bmpman/bmpman.cpp (hunk 1 above)
Comment (The-E) [reply code/bmpman/bmpman.cpp]: I would agree - if something critical happens, like an asset not being loaded, then that should be in the normal log.

Comment (EatThePath) [reply code/bmpman/bmpman.cpp]: Latest pushes hopefully cover both these issues.

**Review by Goober5000 — APPROVED** [review]
Looks good now.

File: code/model/modelread.cpp [hunk 5]
```diff
@@ -3403,7 +3403,7 @@ int model_load(const  char* filename, int n_subsystems, model_subsystem* subsyst
 					dl2--;	// Start from 1 up...
 					if (dl2 >= sm1->num_details ) sm1->num_details = dl2+1;
 					sm1->details[dl2] = j;
-  				    mprintf(( "Submodel '%s' is detail level %d of '%s'\n", sm2->name, dl2 + 1, sm1->name ));
+  				    nprintf(("Model", "Submodel '%s' is detail level %d of '%s'\n", sm2->name, dl2 + 1, sm1->name ));
```
Comment (BMagnu) [inline code/model/modelread.cpp:3406]: Not sure how I feel about this one being hidden.
Accidental LOD-ing is _the_ most common cause for submodels misbehaving I have seen, and it being hidden makes it a lot harder to spot on a glance from logs from non-tech-savvy people.

Comment (Goober5000) [reply code/model/modelread.cpp:3406]: Good point

File: code/bmpman/bmpman.cpp [hunk 6]
```diff
@@ -1600,7 +1600,7 @@ int bm_load_animation(const char *real_filename, int *nframes, int *fps, int *ke
 				throw e;
 			}
 			else {
-				mprintf(("Failed to load apng: %s\n", e.what()));
+				nprintf(("apng","Failed to load apng: %s\n", e.what()));
```
Comment (Goober5000) [inline code/bmpman/bmpman.cpp:1603]: This should be displayed

File: code/graphics/generic.cpp [hunk 7]
```diff
@@ -217,7 +217,7 @@ int generic_anim_stream(generic_anim *ga, const bool cache)
 				ga->png.anim = new apng::apng_ani(ga->filename, cache);
 			}
 			catch (const apng::ApngException& e) {
-				mprintf(("Failed to load apng: %s\n", e.what() ));
+				nprintf(("apng","Failed to load apng: %s\n", e.what() ));
```
Comment (Goober5000) [inline code/graphics/generic.cpp:220]: This should be displayed

File: code/model/modelread.cpp (hunk 5 above)
Comment (Goober5000) [inline code/model/modelread.cpp:3406]: This should be displayed

**Review by Goober5000 — CHANGES_REQUESTED** [review]
On second thought, three more messages should be in the log by default, including the one @BMagnu pointed out

File: code/model/modelread.cpp (hunk 5 above)
Comment (EatThePath) [reply code/model/modelread.cpp:3406]: Alright. I think the message is awkwardly formatted and confusing in isolation, but that's a seperate issue. I will revert.

File: code/graphics/generic.cpp (hunk 7 above)
Comment (EatThePath) [reply code/graphics/generic.cpp:220]: I have a minor vendeta against these apng messages... in my most recent level load in WM I got almost 200 lines of this complaining about *glow maps* that are all valid. This was supposedly fixed after I filed it for #3644, it's the exact smage message, but I can't remember if I verified the fix at the time or if this was just working as intended.

I can restore them I guess, if they're ever actually useful for casual debugging.

**Review by Goober5000 — APPROVED** [review]

**Pull request merged**